src/app/filecoin-plus/allocators/constants/sorting.ts

```diff
@@ -0,0 +1,28 @@
+import type { ColumnSort, SortDirection } from '@tanstack/react-table'
+
+import { findOrThrow } from '@/utils/findOrThrow'
+
+import type { TableSortOption } from '../components/SelectSort'
+
+const NAME_INITIAL_TABLE_STATE = {
+  id: 'name',
+  desc: false,
+} as const satisfies ColumnSort
+
+const DEFAULT_SORT_OPTION_ID = (
+  NAME_INITIAL_TABLE_STATE.desc ? 'desc' : 'asc'
+) satisfies SortDirection
+
+const NAME_SORT_OPTIONS = [
+  { id: 'asc', name: 'Organization (A-Z)' },
+  { id: 'desc', name: 'Organization (Z-A)' },
+] as const satisfies Array<TableSortOption>
+
+export const NAME_SORT_CONFIG = {
+  INITIAL_TABLE_STATE: NAME_INITIAL_TABLE_STATE,
+  OPTIONS: NAME_SORT_OPTIONS,
+  DEFAULT_OPTION: findOrThrow(
+    NAME_SORT_OPTIONS,
+    (option) => option.id === DEFAULT_SORT_OPTION_ID,
+  ),
+} as const
```

Comment:
I'm finding this quite hard to read. And even though I saw your comment about `satisfies` - I'm failing to grasp the usefulness here. Let's Huddle later?

```suggestion
import type { TableSortOption } from '../components/SelectSort'

const nameSortOptions: Array<TableSortOption> = [
  { id: 'asc', name: 'Organization (A-Z)' },
  { id: 'desc', name: 'Organization (Z-A)' },
] as const

export const nameSortConfig = {
  initialTableState: {
    id: 'name',
    desc: false,
  },
  options: nameSortOptions,
  defaultOption: nameSortOptions[0],
} as const
```

Reply:
`satisfies` is a looser way to type values, I did my last skill stretch on it 😅 Anyway, I'm not attached to it, it can go!

`satisfies` aside, what I'm trying to do here is ensure that the default option is the same one as defined in the table's initial state.

Reply:
I reworked it to make it closer to your suggestions. Let me know what you think :)